agent/src/testflinger_agent/agent.py

```diff
@@ -124,66 +127,86 @@ def _post_initial_agent_data(self):
-        agent = self.client.config.get("agent_id")
-        files = [
-            "/tmp/TESTFLINGER-DEVICE-OFFLINE-{}".format(agent),
-            "/tmp/TESTFLINGER-DEVICE-OFFLINE-{}".format(
-                agent.replace("-", "")
-            ),
-        ]
-        return files
-
-    def get_restart_files(self):
-        # Return possible restart filenames with and without dashes
-        # i.e. support both:
-        #     TESTFLINGER-DEVICE-RESTART-devname-001
-        #     TESTFLINGER-DEVICE-RESTART-devname001
-        agent = self.client.config.get("agent_id")
-        files = [
-            "/tmp/TESTFLINGER-DEVICE-RESTART-{}".format(agent),
-            "/tmp/TESTFLINGER-DEVICE-RESTART-{}".format(
-                agent.replace("-", "")
-            ),
-        ]
-        return files
-
-    def check_offline(self):
-        possible_files = self.get_offline_files()
-        for offline_file in possible_files:
-            if os.path.exists(offline_file):
-                self.set_agent_state("offline")
-                return offline_file
-        self.set_agent_state("waiting")
-        return ""
-
-    def check_restart(self):
-        possible_files = self.get_restart_files()
-        for restart_file in possible_files:
-            if os.path.exists(restart_file):
-                try:
-                    os.unlink(restart_file)
-                    logger.info("Restarting agent")
-                    self.set_agent_state("offline")
-                    sys.exit("Restart Requested")
-                except OSError:
-                    logger.error(
-                        "Restart requested, but unable to remove marker file!"
-                    )
-                    break
+    def get_agent_state(self) -> tuple:
+        """Get the agent state from the server by using client module.
+
+        :return: State for the agent and reason for the state if any.
+        """
+        agent_data = self.client.get_agent_data(self.agent_id)
+        # Comment is optional, so key might not exists
+        comment = agent_data.get("comment", "")
+        try:
+            agent_state = agent_data["state"]
+        except KeyError:
+            logger.error(
+                "Unable to retrieve status for agent: %s "
+                "Assumming agent is offline",
+                self.agent_id,
+            )
+            agent_state = AgentState.OFFLINE
+        return (agent_state, comment)
+
+    def check_offline(self) -> tuple:
+        """Determine if agent should be taken offline.
+
+        :return: True or False along with the comment if any.
+        """
+        agent_state, comment = self.get_agent_state()
 
-    def mark_device_offline(self):
-        # Create the offline file, this should work even if it exists
-        open(self.get_offline_files()[0], "w").close()
+        # Offline set by server
+        if (
+            agent_state == AgentState.OFFLINE
+            or agent_state == AgentState.MAINTENANCE
+        ):
```

Comment:
Might need formatting, but this would be easier to manage in case we ever add more "offline" states

```suggestion
        if agent_state in (
            AgentState.OFFLINE,
            AgentState.MAINTENANCE,
        ):
```